model/task/task_annotations_test.go

```diff
@@ -0,0 +1,155 @@
+package task
```

Comment:
Can we add some checks to these tests to make sure that they are setting HasAnnotations as expected? Maybe include some examples of when they shouldn't set it?

Reply:
Sure, added some more checks & cases